src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -207,63 +197,115 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
+            clusterMonitorSupervisor ! ClusterCreated(cluster, clusterRequest.stopAfterCreation.getOrElse(false))
+          case Failure(e) =>
+            logger.info(s"Failed the asynchronous portion of the creation of cluster '$clusterName' " +
+              s"on Google project '${googleProject.value}'.")
+
+            // We also want to record the error in database for future reference.
+            val errorMessage = s"Asynchronous creation of cluster '$clusterName' on Google project" +
+              s"'${googleProject.value}' failed due to '${e.getMessage}'."
+            // TODO Make errorCode field nullable in ClusterErrorComponent and pass None below
+            val dummyErrorCode = -1
+            val errorInfo = ClusterError(errorMessage, dummyErrorCode, Instant.now)
+            dbRef.inTransaction(_.clusterErrorQuery.save(cluster.id, errorInfo))
+
+            // Since we failed, createGoogleCluster removes resources in Google but
+            // we also need to notify our auth provider that the cluster has been deleted.
+            // We won't wait for that deletion, though.
+            authProvider.notifyClusterDeleted(userEmail, userEmail, googleProject, clusterName)
+        }
     }
+
+    attemptToSaveClusterInDb
   }
 
   // Meant to be run asynchronously to the clusterCreate API request
-  def completeClusterCreation(userEmail: WorkbenchEmail,
-                              serviceAccountInfo: ServiceAccountInfo,
-                              googleProject: GoogleProject,
-                              clusterName: ClusterName,
-                              clusterRequest: ClusterRequest): Future[(Cluster, GcsBucketName, Option[ServiceAccountKey])] = {
-    // Notify the AuthProvider
-    authProvider.notifyClusterCreated(userEmail, googleProject, clusterName)
-    // Proceed to creating the cluster on the Google side
-    // TODO Update this method so the stuff it returns is persisted
-    createGoogleCluster(userEmail, serviceAccountInfo, googleProject, clusterName, clusterRequest)
+  private def completeClusterCreation(userEmail: WorkbenchEmail,
+                              cluster: Cluster,
+                              clusterRequest: ClusterRequest): Future[Unit] = {
+    for {
+      _ <- authProvider.notifyClusterCreated(userEmail, cluster.googleProject, cluster.clusterName)
+      _ = logger.info(s"Successfully notified the AuthProvider for creation of cluster ${cluster.clusterName} " +
+        s"on Google project ${cluster.googleProject.value}. Proceeding to creating the cluster on Google Dataproc...")
+
+      (googleCluster, initBucket, serviceAccountKey) <- createGoogleCluster(userEmail, cluster, clusterRequest)
+    } yield {
+      logger.info(s"Successfully finished asynchronously creating the cluster ${cluster.clusterName} " +
+        s"on Google project ${cluster.googleProject.value}. Proceeding to updating the database cluster record...")
+      dbRef.inTransaction {
```

Comment:
This seems wrong to have a `Future` inside a `yield` block. If I'm reading correctly this method would return `Future[Future[Int]]` if it weren't being coerced to `Future[Unit]`. 

Better would be doing the DB update inside the for-comprehension:
```
for {
  _ <- authProvider.notifyClusterCreated(...)
  _ = logger.info("...{)
  _ <- (googleCluster, initBucket, serviceAccountKey) <- createGoogleCluster(...)
  _ <- dbRef.inTransaction { _.clusterQuery. updateAsyncClusterCreationFields(...) }
} yield ()
```